src/main.cpp

```diff
@@ -0,0 +1,1616 @@
+
+    std::string msg_temp; // TODO start using g_msg_temp instead
+    std::string hint_name;
+    std::string hint_value;
+
+    std::int32_t window_x;
+    std::int32_t window_y;
+    std::int32_t window_w;
+    std::int32_t window_h;
+    std::uint32_t window_flags;
+    std::string window_title;
+
+    struct QueueFamilyIndices family_indices;
+
+    SDL_Window* main_window;
+
+    VkApplicationInfo application_info;
+    VkInstanceCreateInfo create_info;
+
+    VkResult result;
+    VkInstance vulkan_instance;
+    VkPhysicalDevice physical_device;
+    VkDevice logical_device;
+    VkQueue drawing_queue;
+    VkQueue presentation_queue;
+    VkSurfaceKHR surface;
+    VkSwapchainKHR swapchain;
+    VkFormat swapchain_image_format;
+    VkExtent2D swapchain_extent;
+    VkRenderPass render_pass;
+    VkPipelineLayout pipeline_layout;
+    VkPipeline graphics_pipeline;
+    VkCommandPool command_pool;
+    VkCommandBuffer command_buffer;
+
+    VkSemaphore semaphore_image_available;
+    VkSemaphore semaphore_render_finished;
+    VkFence fence_in_flight;
+
+    std::vector<VkImageView> swapchain_image_views;
+    std::vector<VkImage> swapchain_images;
+    std::uint32_t swapchain_image_count;
+
+    const std::string application_name = g_program_name;
+    const std::string engine_name = "No engine";
+
+    ret = -1;
+    flags = 0;
+
+    extension_count = 0;
+    extension_names.resize(0);
+    swapchain_framebuffers.resize(0);
+
+    hint_name = SDL_HINT_RENDER_SCALE_QUALITY;
+    hint_value = "1";
+
+    window_x = SDL_WINDOWPOS_UNDEFINED;
+    window_y = SDL_WINDOWPOS_UNDEFINED;
+    window_w = 640;
+    window_h = 480;
+    window_flags = SDL_WINDOW_VULKAN;
+    window_title = g_program_name;
+
+    application_info = {};
+    create_info = {};
+
+    result = VK_ERROR_UNKNOWN;
+    vulkan_instance = {};
+    physical_device = VK_NULL_HANDLE;
+    logical_device = VK_NULL_HANDLE;
+    drawing_queue = VK_NULL_HANDLE;
+    presentation_queue = VK_NULL_HANDLE;
+    surface = VK_NULL_HANDLE;
+    swapchain = VK_NULL_HANDLE;
+    swapchain_image_format = VK_FORMAT_UNDEFINED;
+    swapchain_extent = {};
+    render_pass = {};
+    pipeline_layout = {};
+    graphics_pipeline = {};
+    command_pool = {};
+
+    swapchain_image_views.resize(0);
+    swapchain_images.resize(0);
+    swapchain_image_count = 0;
+
+    family_indices = {0};
+
+    msg_temp = g_program_name;
+    msg_temp += " running";
+    Log::i(msg_temp);
+
+    // SDL init
+    flags = SDL_INIT_VIDEO;
+    ret = SDL_Init(flags);
+    if (ret < 0) {
+        msg_temp = "Failed to initialize SDL: ";
+        msg_temp += SDL_GetError();
+        throw std::runtime_error(msg_temp);
+    }
+
+    // SDL setHint
+    ret = SDL_SetHint(hint_name.c_str(), hint_value.c_str());
+    if (ret < 0) {
+        msg_temp = "Setting SDL hint";
+        msg_temp += hint_name;
+        msg_temp += " failed";
+        Log::w(msg_temp);
+    }
+
+    // SDL imgInit # <- necessary here?
```

Comment:
SDL_image library is currently not used, so please remove the comment.

Reply:
Removed in: https://github.com/pottumuusi/neotetris/pull/1/commits/0596f64d4ae5c59ab4c7ea7dc1d48778759d376c